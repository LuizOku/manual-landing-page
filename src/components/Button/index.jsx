import React from 'react';
import PropTypes from 'prop-types';

import { StyledButton } from './styles.css';

const Button = ({
  background, textColor, width, ...rest
}) => <StyledButton background={background} textColor={textColor} width={width} {...rest} />;

Button.propTypes = {
  background: PropTypes.string,
  textColor: PropTypes.string,
  width: PropTypes.string,
};
Button.defaultProps = {
  background: '#34d1d1',
  textColor: '#ffffff',
  width: 'auto',
};

export default Button;
