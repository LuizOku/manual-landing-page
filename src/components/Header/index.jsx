import React from 'react';

import Symblol from '../../assets/manual-symblol.png';
import { Container } from './styles.css';

const Header = () => (
  <Container data-testid="header">
    <img data-testid="header-symblol" src={Symblol} alt="" />
  </Container>

);

export default Header;
