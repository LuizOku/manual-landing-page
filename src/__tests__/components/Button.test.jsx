import { render, screen } from '@testing-library/react';

import React from 'react';
import { Button } from '../../components';

describe('Testing Button Component', () => {
  it('should be able to show the button component', () => {
    render(<Button />);
    const buttonElement = screen.getByTestId('button');

    expect(buttonElement).toBeInTheDocument();
  });

  it('should be able to match snapshot', () => {
    render(<Button />);
    const buttonElement = screen.getByTestId('button');

    expect(buttonElement).toMatchSnapshot();
  });
});
