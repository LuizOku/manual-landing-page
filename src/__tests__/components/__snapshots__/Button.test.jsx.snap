// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button Component should be able to match snapshot 1`] = `
<button
  class="sc-bdfBwQ cBGQwN"
  data-testid="button"
  width="auto"
/>
`;
